dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -345,7 +348,16 @@ def _update_categories(
         :type df_series: pandas.DataFrame
         :return: None
         """
-        category_count = df_series.value_counts(dropna=False).to_dict()
+        if self._cms_confidence:
+            category_count = defaultdict(int)
+            num_hashes = count_min_sketch.suggest_num_hashes(self._cms_confidence)
+            num_buckets = count_min_sketch.suggest_num_buckets(self._cms_relative_error)
+            cm = count_min_sketch(num_hashes, num_buckets)
```

Comment:
I'd suggest moving this to init of the class above. The `cm` should persist across updates in case we do batches or streaming datasets

Reply:
nice